### DISCUSSION TITLE
Support `CANCEL` event on invoke for promises: `onCancel`

### DISCUSSION THREAD

**Discussion by knownasilya** [statelyai/xstate#1300] [General]
Basically have a wrapper for generator services that handle cancel, but with invoke there is no `onCancel` so cancelations register as `onError`, would love for that to be a thing.

Here's the wrapper: https://github.com/NullVoxPopuli/emberclear/blob/master/client/web/emberclear/app/utils/ember-concurrency.ts#L34

**Comment by davidkpiano** [comment]
An actor (service) can only be started or stopped by the parent service. It can also "stop itself" once it reaches some completion point. Cancelation can be seen as an event sent to the actor, but the details are specific to the implementation of that actor (there is no standard "cancellation" concept).

Is this about supporting `AbortController`?

**Comment by knownasilya** [comment]
Yes and no, http://ember-concurrency.com/docs/tutorial/refactor

But the same would be true of an abstraction on top of abort controller.
